src/map/utils/battleutils.cpp

```diff
@@ -730,6 +730,16 @@ namespace battleutils
 
                         if (PDefender->isAlive())
                         {
+                            // Subpower is the remaining damage that can be drained. When it reaches 0 the effect ends
+                            int remainingDrain = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_DREAD_SPIKES)->GetSubPower();

```

Comment:
Would you might storing the status effect in a local variable. That'll make this easier to read. It can then be used in both places.
`CStatusEffect* PStatus = ..`